libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -439,6 +439,23 @@ DeclarationRegistrationHelper::DeclarationRegistrationHelper(
 	solAssert(m_currentScope == _currentScope, "Scopes not correctly closed.");
 }
 
+bool DeclarationRegistrationHelper::isOverloadedFunction(
+	Declaration const& _declaration1,
+	Declaration const& _declaration2
+)
+{
+	auto const* function1 = dynamic_cast<FunctionDefinition const*>(&_declaration1);
+	auto const* function2 = dynamic_cast<FunctionDefinition const*>(&_declaration2);
+
+	if (!function1 || !function2)
+		return false;
+
+	if (function1->parameters() != function2->parameters())
```

Comment:
This comparison compares vectors of AST nodes. They will be different if the functions are different.